sdk/src/test/scala/coop/rchain/sdk/merging/DagMergingLogicSpec.scala

```diff
@@ -0,0 +1,170 @@
+  }
+
+  // some random conflict maps and rejection options, computed manually
+  "rejection options" should "be computed correctly" in {
+    computeRejectionOptions(
+      Map(
+        1 -> Set(2, 3, 4),
+        2 -> Set(1),
+        3 -> Set(1, 2),
+        4 -> Set(1)
+      )
+    ) shouldBe Set(Set(1, 2), Set(2, 3, 4))
+
+    computeRejectionOptions(
+      Map(
+        1 -> Set(2, 3, 4),
+        2 -> Set(1, 3, 4),
+        3 -> Set(1, 2, 4),
+        4 -> Set(1, 2, 3)
+      )
+    ) shouldBe Set(Set(2, 3, 4), Set(1, 3, 4), Set(1, 2, 4), Set(1, 2, 3))
+
+    computeRejectionOptions(
+      Map(
+        1 -> Set(2, 3, 4),
+        2 -> Set(1),
+        3 -> Set(1, 4),
+        4 -> Set(1, 3)
+      )
+    ) shouldBe Set(Set(2, 3, 4), Set(1, 3), Set(1, 4))
+
+    computeRejectionOptions(
+      Map(
+        1 -> Set.empty[Int],
+        2 -> Set(3),
+        3 -> Set(2, 4),
+        4 -> Set(3)
+      )
+    ) shouldBe Set(Set(3), Set(2, 4))
+
+    val all          = (1 to 1000) toSet
+    val conflictsMap = (1 to 1000).map(i => i -> (all - i)).toMap
+    computeRejectionOptions(conflictsMap) shouldBe (1 to 1000).map(all - _).toSet
+  }
+
+  "computeOptimalRejection" should "pick rejection with minimal cost. " +
+    "Equal cost should be resolved by size and minimal element" in {
+    val rejectonOptions = Set(Set(1, 2, 3), Set(2, 3, 4), Set(1, 2), Set(2), Set(1))
+    val cost            = Set(1, 2, 3, 4).map(_ -> 1L).toMap
+    computeOptimalRejection(rejectonOptions, cost(_: Int)) shouldBe Set(1)
+  }
+
+  "computeMergeableOverflowRejectionOptions" should "add to rejection options items leading to overflow." in {
+    val conflictSet         = Set(1, 2, 3, 4)
+    val rejectOptions       = Set(Set(1), Set(2))
+    val initMergeableValues = Map("a" -> 0L, "b" -> 2L)
+    // 2 should be added to rejection options as it goes negative on channel "a", 3 does it on channel "b"
+    // 4 passes in as on channel "b" it does 2 - 1
+    val mergeableDiffs = Map(2 -> Map("a" -> -1L), 3 -> Map("b" -> -3L), 4 -> Map("b" -> -1L))
+    addMergeableOverflowRejections[Int, String](
+      conflictSet,
+      rejectOptions,
+      initMergeableValues,
+      mergeableDiffs
+    ) shouldBe Set(Set(1, 2, 3), Set(2, 3))
+  }
+
+  "computeMergeableOverflowRejectionOptions" should "sort deploys by sum of absolute diffs to fold mergeable value" in {
+    val conflictSet         = Set(1, 2, 3, 4, 5)
+    val rejectOptions       = Set.empty[Set[Int]]
+    val initMergeableValues = Map("a" -> 0L)
+    val mergeableDiffs = Map(
+      1 -> Map("a" -> 10L),
+      2 -> Map("a" -> -5L), // this is applied first and immediately goes negative, so rejected
+      3 -> Map("a" -> 15L),
+      4 -> Map("a" -> 10L),
+      5 -> Map("a" -> -20L)
+    )
+    addMergeableOverflowRejections[Int, String](
+      conflictSet,
+      rejectOptions,
+      initMergeableValues,
+      mergeableDiffs
+    ) shouldBe Set(Set(2))
+  }
+}
```

Comment:
This type of test is equivalent to <https://github.com/rchain/rchain/blob/ed9ffb16fb091a723ed9de6146dc4bb3151dc7fc/casper/src/test/scala/coop/rchain/casper/merging/MergeNumberChannelSpec.scala> without necessity to build rholang state.